docs/source/Mux.rst

```diff
@@ -1,10 +1,10 @@
-.. _mux:
+.. _variants:
 
-===================
-Test variants - Mux
-===================
+========================
+Test variants - Variants
```

Comment:
If you remember I asked you to wait with the nice and shiny documentation at the end of the modification, because the bases of `Multiplexer` are changing and as it's already outdated, changing it per each part wouldn't be beneficial (I'd have to fix the current issues, than define terms for this version and then rewrite it from scratch for the plugin-based version).

The purpose of this documentation change was to briefly sync the changes in names to avoid big confusion, but the purpose is not to provide extensive description of the Varianter internals. The PR with turning `Varianter` into a pluggable interface should contain the up-to-date documentation.